lucas_scheinkerman/ROS_investigation.md

```diff
@@ -0,0 +1,222 @@
+ /gazebo/set_link_state [gazebo_msgs/LinkState]
+ /tf [tf2_msgs/TFMessage] 3 subscribers
+ /create1/move_base_simple/goal [geometry_msgs/PoseStamped]
+ /create1/mobile_base/sensors/bumper_pointcloud [sensor_msgs/PointCloud2]
+ /create1/joint_states [sensor_msgs/JointState]
+ /create1/cmd_vel [geometry_msgs/Twist]
+ /create1/move_base/local_costmap/costmap_updates [map_msgs/OccupancyGridUpdate]
+ /create2/move_base/local_costmap/costmap_updates [map_msgs/OccupancyGridUpdate]
+ /tf_static [tf2_msgs/TFMessage] 3 subscribers
+ /create1/move_base/global_costmap/footprint [geometry_msgs/PolygonStamped]
+ /map_updates [map_msgs/OccupancyGridUpdate]
+ /map [nav_msgs/OccupancyGrid] 3 subscribers
+ /rosout [rosgraph_msgs/Log]
+ /create1/gts [nav_msgs/Odometry]
+ /create1/initialpose [geometry_msgs/PoseWithCovarianceStamped]
+ /create1/odom [nav_msgs/Odometry]
+ /create1/nodelet_manager/bond [bond/Status] 2 subscribers
+ /gazebo/set_model_state [gazebo_msgs/ModelState]
+ /create1/move_base/goal [move_base_msgs/MoveBaseActionGoal]
+ /create1/cliff [ca_msgs/Cliff]
+ /create2/move_base/local_costmap/costmap [nav_msgs/OccupancyGrid]
+ /clock [rosgraph_msgs/Clock] 12 subscribers
+ /create1/move_base/local_costmap/costmap [nav_msgs/OccupancyGrid]
+ /create1/bumper [ca_msgs/Bumper]
+ /create1/laser/scan [sensor_msgs/LaserScan]
+```
+
+## Listening to a specific topic
+
+This assignment was completed by executing the command 'rostopic echo /create1/cmd_vel', which corresponds to the topic that has information about the angular and linear velocity of the simulated robot. Simultaneously with the printing made by this command, the robot was being moved with the 'keyboard_teleop' tool. The result was:
+
+```console
+create@galatea:/create_ws$ rostopic echo /create1/cmd_vel
+linear:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+---
+linear:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+---
+linear:
+  x: 0.3
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+---
+linear:
+  x: 0.3
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: -0.8
+---
+linear:
+  x: 0.0
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: -0.8
+---
+linear:
+  x: -0.3
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: -0.8
+---
+linear:
+  x: -0.3
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: -0.8
+```
+
+## Using rqt_graph with a quick comment about the 'Multi-robot setup (#76)' commit
+
+The commit mentioned in the last title added another feature to the package which allows controlling multiple robot in one simulation. This creates a bug related to the namespaces of the different robots and the 'keyboard_teleop' tool. This tool sends directly messages to the 'cmd_vel' topic, but this topic doesn't exist. The 'cmd_vel' topic that is available is '/create1/cmd_vel'
+This can be seen in the following image:
+
+![rqt_graph without namespace modification](media/screen_without_ns.png)
```

Comment:
Replace the images showing the topic names too. And highlight the nodes and topics of interest.